types/wlr_keyboard_shortcuts_inhibit_v1.c

```diff
@@ -0,0 +1,228 @@
+			ZWP_KEYBOARD_SHORTCUTS_INHIBIT_MANAGER_V1_ERROR_ALREADY_INHIBITED,
+			"this surface already has keyboard shortcuts "
+			"inhibited on this seat");
+		return;
+	}
+
+	struct wlr_keyboard_shortcuts_inhibitor_v1 *inhibitor =
+		calloc(1, sizeof(struct wlr_keyboard_shortcuts_inhibitor_v1));
+	if (!inhibitor) {
+		wl_client_post_no_memory(client);
+		return;
+	}
+
+	struct wl_resource *wl_resource = wl_resource_create(client,
+		&zwp_keyboard_shortcuts_inhibitor_v1_interface, 1, id);
+	if (!wl_resource) {
+		wl_client_post_no_memory(client);
+		free(inhibitor);
+		return;
+	}
+
+	inhibitor->resource = wl_resource;
+	inhibitor->surface = surface;
+	inhibitor->seat_client = seat_client;
+	inhibitor->active = false;
+	wl_signal_init(&inhibitor->events.destroy);
+
+	inhibitor->surface_destroy.notify =
+		keyboard_shortcuts_inhibitor_handle_surface_destroy;
+	wl_signal_add(&surface->events.destroy, &inhibitor->surface_destroy);
+
+	wl_resource_set_implementation(wl_resource,
+		&keyboard_shortcuts_inhibitor_impl, inhibitor,
+		keyboard_shortcuts_inhibitor_v1_handle_resource_destroy);
+
+	wl_list_insert(&manager->inhibitors, &inhibitor->manager_link);
+	wl_list_insert(&surface->keyboard_shortcuts_inhibitors, &inhibitor->surface_link);
+	wlr_signal_emit_safe(&manager->events.new_inhibitor, inhibitor);
+}
+
+static void manager_handle_destroy(struct wl_client *client,
+		struct wl_resource *manager_resource) {
+	wl_resource_destroy(manager_resource);
+}
+
+static const struct zwp_keyboard_shortcuts_inhibit_manager_v1_interface
+keyboard_shortcuts_inhibit_impl = {
+	.destroy = manager_handle_destroy,
+	.inhibit_shortcuts = manager_handle_inhibit_shortcuts,
+};
+
+static void handle_display_destroy(struct wl_listener *listener, void *data) {
+	struct wlr_keyboard_shortcuts_inhibit_manager_v1
+		*keyboard_shortcuts_inhibit = wl_container_of(listener,
+			keyboard_shortcuts_inhibit, display_destroy);
+	wlr_signal_emit_safe(&keyboard_shortcuts_inhibit->events.destroy,
+		keyboard_shortcuts_inhibit);
+	wl_list_remove(&keyboard_shortcuts_inhibit->display_destroy.link);
+	wl_global_destroy(keyboard_shortcuts_inhibit->global);
+	free(keyboard_shortcuts_inhibit);
+}
+
+static void keyboard_shortcuts_inhibit_bind(struct wl_client *wl_client,
+		void *data, uint32_t version, uint32_t id) {
+	struct wlr_keyboard_shortcuts_inhibit_manager_v1
+		*keyboard_shortcuts_inhibit = data;
+
+	struct wl_resource *wl_resource = wl_resource_create(wl_client,
+		&zwp_keyboard_shortcuts_inhibit_manager_v1_interface,
+		version, id);
+	if (!wl_resource) {
+		wl_client_post_no_memory(wl_client);
+		return;
+	}
+
+	wl_resource_set_implementation(wl_resource,
+		&keyboard_shortcuts_inhibit_impl,
+		keyboard_shortcuts_inhibit, NULL);
+}
+
+struct wlr_keyboard_shortcuts_inhibit_manager_v1 *
+wlr_keyboard_shortcuts_inhibit_v1_create(struct wl_display *display) {
+	struct wlr_keyboard_shortcuts_inhibit_manager_v1
+		*keyboard_shortcuts_inhibit = calloc(1,
```

Comment:
Style: we generally break this kind of lines right after the `=` sign

Reply:
Done (assuming I caught them all).